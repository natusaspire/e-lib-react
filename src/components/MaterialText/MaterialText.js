import React, { memo } from 'react';

const MaterialText = ({ material }) => (
  <div className="card">
    <div className="card__header">{material.title}</div>
    <div className="content">
      <span>{material.textContent}</span>
    </div>
  </div>
);

export default memo(MaterialText);
