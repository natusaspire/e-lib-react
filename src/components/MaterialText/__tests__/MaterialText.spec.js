import React from 'react';
import { shallow } from 'enzyme';

import MaterialText from '@/components/MaterialText';

describe('<MaterialText />', () => {
  it('should render text material', () => {
    const material = {
      id: 1,
      type: 'text',
      title: 'Material title',
      textContent: 'Material text content.'
    };

    const wrapper = shallow(<MaterialText material={material} />);

    expect(wrapper).toMatchSnapshot();
  });
});
