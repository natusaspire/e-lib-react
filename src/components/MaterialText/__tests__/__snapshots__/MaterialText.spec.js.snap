// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaterialText /> should render text material 1`] = `
<div
  className="card"
>
  <div
    className="card__header"
  >
    Material title
  </div>
  <div
    className="content"
  >
    <span>
      Material text content.
    </span>
  </div>
</div>
`;
